mlir/lib/Dialect/Rock/Transforms/MakeGepsInbounds.cpp

```diff
@@ -0,0 +1,64 @@
+//===- CleanMath.cpp - Clean up math after lowering/unrolling loops  ---===//
+//
+// Copyright 2022 AMD
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Dialect/Arith/Transforms/Passes.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/Rock/Passes.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/Pass/PassManager.h"
+#include "mlir/Transforms/Passes.h"
+
+// Remove these includes after we're not working with a test pass upstream
+#include "mlir/Analysis/DataFlow/DeadCodeAnalysis.h"
+#include "mlir/Analysis/DataFlow/IntegerRangeAnalysis.h"
+#include "mlir/Transforms/FoldUtils.h"
+
+
+namespace mlir {
+namespace rock {
+// TODO: ask why?
```

Comment:
Magic incantations that let us autogenerate a _lot_ of boilerplate.

Reply:
Forgot to delete) Thanks for answering!